### ISSUE TITLE
Future: King of Fighters 97 Global Match - Steam

### ISSUE THREAD

**Issue by fliperama86** [shawngmc/game-extraction-toolbox#69]
It looks like Steam's "King of Fighters 97 Global Match" is pretty close to KOF97 Mame rom. Comparing the files from these versions, here is what I was able to find (`[Steam game directory]\Data\rom` vs `kof97.zip`):

| Steam | Mame | CRC | Match |
|---|---|---|---|
| c1.bin | ? | E9CB47E7 | ❌ |
| m1.bin | 232-m1.m1 | 45348747 | ✅ |
| p1.bin (First 1.048.576 bytes) | 232-p1.p1 | 7DB81AD9 | ✅ |
| p1.bin (Other 4.194.304.bytes) | 232-p2.sp2 | 158B23F6 | ✅ |
| s1.bin <br /> s2.bin | 232-s1.s1 | 8514ECF5 | ✅ |
| sp_4j.bin JP Bios | N/A | DFF6D41F |  |
| sp_4s.bin (US Bios?) | N/A | 142E97CC |   |
| v1.bin split in 3 (1/3) | 232-v1.v1 | 22A2B5B5 | ✅ |
| v1.bin split in 3 (2/3) | 232-v1.v2 | 2304E744 | ✅ |
| v1.bin split in 3 (3/3) | 232-v1.v3 | 759EB954 | ✅ |

`c1.bin` _seems_ to contain the sprite rom data, "_equivalent_" to `.c1` to `.c6` on `kof97.zip`, since they all add up to exactly 40MB, but they are **not** equal and the CRCs don't match 😞

By splitting `c1.bin` in 6 parts (4x 8MB + 2x 4MB) and zipping all files from the Steam version together - and properly renamed - I managed to boot the rom in RetroArch's Mame. The problem is, most of the sprites are displayed incorrectly:

<details>
  <summary>Screenshots</summary>

![kof97-231123-123201](https://github.com/shawngmc/game-extraction-toolbox/assets/151855083/95b21408-e9f9-4413-a2c2-2fe61f52d9cc)

![kof97-231123-125539](https://github.com/shawngmc/game-extraction-toolbox/assets/151855083/c1931a09-e531-4f4b-9dfe-791e173c4d79)

</details>

Again, it seems like the only missing piece is figuring out how to fix the sprite rom data in `c1.bin`.

**Comment by fliperdeboteco** [comment]
Interestingly enough, by running the built (from Steam files) `kof97.zip` against the tool [NeoBuilder Homebrew GUI v1.06](https://bidouillouzzz.blogspot.com/2022/02/neobuilder-homebrew-gui.html) I was able to create a `.neo` file that worked on my Mister FPGA, which was my ultimate goal 🎉

**Comment by shawngmc** [comment]
Hmm... that implies that the ROM data is there in full, and just requires
transformations. When I get a chance, I'll take a look.

**Comment by RealRelativeEase** [comment]
There's a bash script to extract a working ROM of KOF'97 from the Amazon version of Global Match: [https://github.com/RedundantCich/goNCommand/blob/main/GoKOF97](https://github.com/RedundantCich/goNCommand/blob/main/GoKOF97)

Maybe this could be of use for the Steam version as well?

**Comment by fliperdeboteco** [comment]
> There's a bash script to extract a working ROM of KOF'97 from the Amazon version of Global Match: https://github.com/RedundantCich/goNCommand/blob/main/GoKOF97
> 
> Maybe this could be of use for the Steam version as well?

Yep it worked! I will create a PR for this repo to include a command for this game.
Thank you!!!

**Comment by RealRelativeEase** [comment]
Credit goes to lioneltrs and RedundantCich, of course. Still glad I was able to be of service!

**Comment by RealRelativeEase** [comment]
Maybe the same can also be done with the other Neo Geo ports developed by Code Mystics?

- Garou: Mark of the Wolves
- Samurai Shodown V Special
- The Last Blade 2 (there's already a [bash script](https://github.com/lioneltrs/goNCommand/blob/main/goLB2) for it)

The King of Fighters '98: Ultimate Match also contains a playable AES version of the Neo Geo original, but the file structure is different compared to other Code Mystics ports. So it might not be a full ROM or something else.

**Comment by fliperdeboteco** [comment]
@RealRelativeEase 
Perhaps. I don't own Garou: Mark of the Wolves, Samurai Shodown V Special nor The Last Blade 2 on Steam, besides the first two are already already extractable from the Humble Bundle version.
I took a look at King of Fighters '98: Ultimate Match and it doesn't look like it is emulated at all. Did you find any reasons to think otherwise?

**Comment by RealRelativeEase** [comment]
It was just a wild guess that the AES version of KOF'98 inside of Ultimate Match is ROM-based, since I couldn't spot any differences compared to a regular ROM of the game run in AES mode. Aside from the option to return to Ultimate Match in the main menu, which could've been added in there via a patch or a custom hack.

I don't think there were too many full-on ports of Neo Geo games past the PS1/Dreamcast, since most of the collections and re-releases seem to contain ROMs wrapped in an emulator.

**Comment by Feilakas** [comment]
As far as I can tell, The KOF '98 UM FE is a port, not a ROM-based release.

I can confirm however that, The KOF '97 GM as well as [The Last Blade 2](https://github.com/shawngmc/game-extraction-toolbox/issues/39#issuecomment-1335413078) are indeed ROM-based releases and both can be turned into usable ROMs.
By following the commits [fliperdeboteco](https://github.com/shawngmc/game-extraction-toolbox/pull/70) created as well as [lioneltrs](https://github.com/lioneltrs/goNCommand)' original bash script, I was able to recreate a stand-alone python script that converts the ROM files into the KOF97.zip and test that it indeed works correctly.

I cannot comment on the"Samurai Shodown V Special" because I instead purchased the Samurai Shodown NeoGeo Collection[https://github.com/shawngmc/game-extraction-toolbox/issues/63](url) released two years later and from those, [ValadAmoleo](https://github.com/ValadAmoleo/sf30ac-extractor) had managed to extract games 1-4 and then [in the following guide](https://milkchoco-info.translate.goog/archives/3705?_x_tr_sl=auto&_x_tr_tl=en&_x_tr_hl=ja&_x_tr_pto=wapp&_x_tr_hist=true), Author alhumbra (Richard Roe) explained how to also extract SS 5S FE. [The following tool]( https://github.com/Feilakas/samsho5_fe) uses said instructions to extract the SS 5V FE ROM.

Finally, I cannot comment on Garou: Mark of the Wolves because I too bought the [Humble Bundle NeoGeo Classic Collection](https://www.humblebundle.com/store/neogeo-classic-complete-collection) which includes this game among others and in that collection, [all ROMs can be easily found in the installer files](https://github.com/shawngmc/game-extraction-toolbox/issues/39).
